os_brick/initiator/connectors/lightos.py

```diff
@@ -62,6 +62,23 @@ def __init__(self,
         self.message_queue = message_queue
         self.DISCOVERY_DIR_PATH = '/etc/discovery-client/discovery.d/'
 
+    @staticmethod
+    def get_ip_addresses():
+        ip_addresses = []
+        result = subprocess.check_output(['ip', 'addr']).decode('utf-8')
```

Comment:
I am not sure this will work, unfortunately. 
1. What guarantees do you have that `ip(1)` is installed? Does openstack require this package? If not, it needs to be added as a requirement, and the openstack maintainers may not agree to add it.
2. The openstack security model is annoying and we had to jump through a lot of hoops to do some things. Look at the `priv_lightos` stuff to see how we dealt with it. This may or may not be relevant here, but since the user can use `ip` to change the system IP address, I wouldn't be terribly surprised if openstack stops you from calling it without elevated privileges.
3. What is the format of `ip addr` output changes? Using binaries in scripts is dangerous when you want a long term solution. Better to use fixed system APIs, if possible.

Reply:
I think all modern Linux OS(s) are pre-packaged with command [ ip addr ] , but now i'm hesitant after you mentioned the security concerns.  Can you rephrase you question in point # 3     **What is the format of ip addr output changes?** 

I'm new to the system API.  Can you explain this     **Better to use fixed system APIs, if possible.**

@muliby-lb

Reply:
Sorry, (3) should have been "what *if* the format changes". I am not sure it's a real concern but at the same time, getting the IP addresses from python, which calls the system APIs, seems pretty simple:

```
import socket

def get_system_ip_addresses():
    ip_addresses = []
    try:
        hostname = socket.gethostname()
        ip_addresses.append(socket.gethostbyname(hostname))
        ip_addresses.extend(socket.gethostbyname_ex(hostname)[-1])
    except Exception as e:
        print(f"An error occurred: {e}")
    return ip_addresses

if __name__ == "__main__":
    system_ip_addresses = get_system_ip_addresses()
    print("System IP addresses:", system_ip_addresses)
```

Reply:
I had tried this approach  but this was giving very limited IPs   e.g  these IPs on my machine 
['127.0.1.1', '127.0.1.1'] 

Then i switched to use [ ip add] which gives all available network ips 
['127.0.0.1', '10.0.0.115', '172.17.0.1', '172.18.0.1', '10.212.1.4']

IP-add  return both IPv4 and IPv6 address.

Reply:
yea - this probably a problem, we somehow check the discovery client is running. I will check how we do that

Reply:
@rahman-lb here's a better way to do what we need in pure python using the `netifaces` package:
https://stackoverflow.com/a/274644

Reply:
I see that in connector.py there is a my_ip
wrote checking if that IP is good for us.

alternatives:
1. if we want to use ip addr its need to be in the privileged module 
2. see how the python way gives the correct addr
3. upload a patch to upstream and see of someone gives a better option

Reply:
Also see this:
https://github.com/openstack/oslo.utils/blob/cecf061e6e4f666104170e3bba93fdae24255309/oslo_utils/netutils.py#L369

you can import it and use it like we do here:
https://github.com/LightBitsLabs/os-brick/blob/7dd8e0fdb828e5020f5c67a3e727e25990125a4b/os_brick/privileged/lightos.py#L18